roles/docker/tasks/install.yml

```diff
@@ -20,17 +20,33 @@
   when: inventory_hostname in groups['gnbsim_nodes']
   become: true
 
-- name: add docker GPG apt Key
-  apt_key:
+- name: Create directory for Docker's GPG key
+  file:
+    path: /etc/apt/keyrings
+    state: directory
+    mode: '0755'
+  when: inventory_hostname in groups['gnbsim_nodes']
+  become: true
+
+- name: Download Docker GPG key
+  get_url:
     url: https://download.docker.com/linux/ubuntu/gpg
-    state: present
+    dest: /etc/apt/keyrings/docker.asc
+    mode: '0644'
+    owner: root
+    group: root
   when: inventory_hostname in groups['gnbsim_nodes']
   become: true
 
-- name: add docker Repository
+- name: Add Docker repository to APT sources
+  vars:
+    arch_mapping:
+      x86_64: amd64
+      aarch64: arm64
   apt_repository:
-    repo: deb https://download.docker.com/linux/ubuntu {{ ansible_distribution_release }} stable
+    repo: "deb [arch={{ arch_mapping[ansible_architecture] | default(ansible_architecture) }} signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu {{ ansible_lsb.codename }} stable"
     state: present
+    filename: docker
```

Comment:
Should not you skip these tasks if `Docker` repository is already present in the host? or how are you handling the case when the host already has the key and repository added to `apt` (e.g., the user installed it)?

Reply:
ansible's `apt_repository` module takes care of it.